PilotDataPlatform/dataops
# PILOT-1880/1980 Add new file status update endpoints

## Summary

- Add two new `task-stream` endpoints for managing file status updates in Redis.
    - These will eventually replace the old `task-dispatch` endpoints.
    - `POST` endpoint creates a new file status entry in Redis Streams.
    - `GET` endpoint retrieves entries from Redis Streams and returns them via an SSE connection.
- Update the local Docker setup to be easier to use.

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1880
https://indocconsortium.atlassian.net/browse/PILOT-1980

## Type of Change


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

1. Review the 2 new endpoints in `dataops/routes/api_task_stream/task_stream.py`.
    1. Are these sufficient for writing all upload, download, copy, and delete records to Redis?
    2. Can these completely replace the `task-dispatch` endpoints? Or is there some functionality missing?
    3. Is there any reason that we would need update/delete endpoints right now? My thoughts are: update is not needed because we'll append new records instead, and delete can come later when we figure out how to auto-trim old entries. But I know these exist in `task-dispatch`.
4. Review the old code in `dataops/routes/api_file_operations/copy_dispatcher.py` and `.../delete_dispatcher.py`.
    1. I didn't touch these. Should I?
    2. There are Redis records being written in there. Do these need to be updated to write to Streams?
